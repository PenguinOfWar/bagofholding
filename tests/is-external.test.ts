import isExternal from '@/packages/is-external/is-external';

it('returns true for an external link', () => {
  const checkLink = isExternal('http://google.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for an external HTTPS link', () => {
  const checkLink = isExternal('https://google.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for an ftp link', () => {
  const checkLink = isExternal('ftp://google.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for an ftps link', () => {
  const checkLink = isExternal('ftps://google.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for an www. link', () => {
  const checkLink = isExternal('www.google.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for an .com .net and .co.uk link and false without', () => {
  const tests = [
    'google.com',
    'google.co.uk',
    'plus.net',
    'gov.uk',
    'pro.sony',
    'japantimes.co.jp'
  ];

  tests.map(test => {
    const checkLink = isExternal(test);
    const checkLinkFalse = isExternal(test.replace(/\./g, ''));

    expect(checkLink).toBeTruthy();
    expect(checkLinkFalse).toBeFalsy();

    expect([checkLink, checkLinkFalse]).toMatchSnapshot();
    return test;
  });
});

it('returns true for a mailto', () => {
  const checkLink = isExternal('mailto:darryl.walker@sony.com');

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns true for a link containing /pdf', () => {
  const checkLink = isExternal('/en_GB/pdf/vpl-ch370', to => {
    return to.toLowerCase().includes('/pdf');
  });

  expect(checkLink).toBeTruthy();
  expect(checkLink).toMatchSnapshot();
});

it('returns false for a relative link', () => {
  const checkLink = isExternal('/en_GB/products/lamp-projectors/vpl-ch370');

  expect(checkLink).toBeFalsy();
  expect(checkLink).toMatchSnapshot();
});
