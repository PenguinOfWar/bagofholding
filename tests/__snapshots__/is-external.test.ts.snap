// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns false for a relative link 1`] = `false`;

exports[`returns true for a link containing /pdf 1`] = `true`;

exports[`returns true for a mailto 1`] = `true`;

exports[`returns true for an .com .net and .co.uk link and false without 1`] = `
[
  true,
  false,
]
`;

exports[`returns true for an .com .net and .co.uk link and false without 2`] = `
[
  true,
  false,
]
`;

exports[`returns true for an .com .net and .co.uk link and false without 3`] = `
[
  true,
  false,
]
`;

exports[`returns true for an .com .net and .co.uk link and false without 4`] = `
[
  true,
  false,
]
`;

exports[`returns true for an .com .net and .co.uk link and false without 5`] = `
[
  true,
  false,
]
`;

exports[`returns true for an .com .net and .co.uk link and false without 6`] = `
[
  true,
  false,
]
`;

exports[`returns true for an external HTTPS link 1`] = `true`;

exports[`returns true for an external link 1`] = `true`;

exports[`returns true for an ftp link 1`] = `true`;

exports[`returns true for an ftps link 1`] = `true`;

exports[`returns true for an www. link 1`] = `true`;
